### PR TITLE
[ALLUXIO-2449] Improve concurrency for getFileInfo

### PR DESCRIPTION
**Pull request by gpang**
https://alluxio.atlassian.net/browse/ALLUXIO-2449

### CODE REVIEW

**Comment by gpang** [comment]
alluxio-bot, check this please

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java [hunk 1]
```diff
@@ -704,6 +704,17 @@ public FileInfo getFileInfo(AlluxioURI path)
       throws FileDoesNotExistException, InvalidPathException, AccessControlException {
     Metrics.GET_FILE_INFO_OPS.inc();
     long flushCounter = AsyncJournalWriter.INVALID_FLUSH_COUNTER;
+
+    // Get a READ lock first to see if we need to load metadata, note that this assumes load
```
Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: isn't the default for load metadata for direct children to load it once?

Comment (gpang) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: On line 772, we use `LoadMetadataOptions.defaults()`, and for that data structure, the `mLoadDirectChildren` field is `false`.

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**